src/elements/element.rectangle.js

```diff
@@ -50,42 +50,76 @@ module.exports = function(Chart) {
 	}
 
 	Chart.elements.Rectangle = Chart.Element.extend({
-		draw: function() {
+		draw: function(isHorizontalBar) {
+
 			var ctx = this._chart.ctx;
 			var vm = this._view;
-
-			var halfWidth = vm.width / 2,
-				leftX = vm.x - halfWidth,
-				rightX = vm.x + halfWidth,
-				top = vm.base - (vm.base - vm.y),
-				halfStroke = vm.borderWidth / 2;
+			var x1,	x2,	y1,	y2;
+			var sign = {};
+			var borderWidth = vm.borderWidth;
+			var borderSkipped = (vm.borderSkipped !== undefined)? vm.borderSkipped: 'bottom';
+
+			if ((isHorizontalBar === undefined) || (!isHorizontalBar)) {
+				// bar
+				x1 = vm.x - vm.width / 2;
+				x2 = vm.x + vm.width / 2;
+				y1 = vm.y;
+				y2 = vm.base;
+				sign.x = 1;
+				sign.y = (y2 > y1) ? 1:-1;
+			} else {
+				// horizontal bar
+				x1 = vm.base;
+				x2 = vm.x;
+				y1 = vm.y - vm.height /2;
+				y2 = vm.y + vm.height /2;
+				sign.x = (x2 > x1) ? 1:-1;
+				sign.y = 1;
+			}
 
 			// Canvas doesn't allow us to stroke inside the width so we can
 			// adjust the sizes to fit if we're setting a stroke on the line
-			if (vm.borderWidth) {
-				leftX += halfStroke;
-				rightX -= halfStroke;
-				top += halfStroke;
+			if (borderWidth) {
+				// borderWidth shold be less than bar width and bar height.
+				var barSize = Math.min(Math.abs(x1 - x2), Math.abs(y1 - y2));
+				borderWidth = (borderWidth > barSize) ? barSize: borderWidth;
+				var halfStroke = borderWidth / 2;
+				// Adjust borderWidth when bar top position is near vm.base(zero).
+				var Temp = {};
```

Comment:
`temp*` is totally fine, however a more descriptive name would be better: not sure what these variables represent right now? maybe `borderLeft` instead of `tempX1`, etc.?

Also please declare these using multiple vars:

```javascript
var tempX1 = x1 + ((borderSkipped !== 'left')? halfStroke * signX: 0);
var tempX2 = x2 + ((borderSkipped !== 'right')? -halfStroke * signX: 0);
var tempY1 = y1 + ((borderSkipped !== 'top')? halfStroke * signY: 0);
var tempY2 = y2 + ((borderSkipped !== 'bottom')? -halfStroke * signY: 0);
```